src/scenic/syntax/compiler.py

```diff
@@ -233,22 +233,77 @@ def visit_Param(self, node: s.Param):
 
     def visit_Require(self, node: s.Require):
         condition = self.visit(node.cond)
-        syntax_id = self._register_requirement_syntax(condition)
+        return self.createRequirementLike(
+            "require", condition, node.lineno, node.name, node.prob
+        )
+
+    def visit_RequireAlways(self, node: s.RequireAlways):
+        condition = self.visit(node.cond)
+        return self.createRequirementLike(
+            "require_always", condition, node.lineno, node.name
+        )
+
+    def visit_RequireEventually(self, node: s.RequireEventually):
+        condition = self.visit(node.cond)
+        return self.createRequirementLike(
+            "require_eventually", condition, node.lineno, node.name
+        )
+
+    def visit_Record(self, node: s.Record):
+        value = self.visit(node.value)
+        return self.createRequirementLike("record", value, node.lineno, node.name)
 
+    def visit_RecordInitial(self, node: s.RecordInitial):
+        value = self.visit(node.value)
+        return self.createRequirementLike(
+            "record_initial", value, node.lineno, node.name
+        )
+
+    def visit_RecordFinal(self, node: s.RecordFinal):
+        value = self.visit(node.value)
+        return self.createRequirementLike("record_final", value, node.lineno, node.name)
+
+    def visit_TerminateWhen(self, node: s.TerminateWhen):
+        condition = self.visit(node.cond)
+        return self.createRequirementLike(
+            "terminate_when", condition, node.lineno, None
+        )
+
+    def visit_TerminateSimulationWhen(self, node: s.TerminateSimulationWhen):
+        condition = self.visit(node.cond)
+        return self.createRequirementLike(
+            "terminate_simulation_when", condition, node.lineno
+        )
+
+    def createRequirementLike(
+        self,
+        functionName: str,
+        syntax: ast.AST,
+        lineno: int,
+        name: Optional[str] = None,
+        prob: Optional[float] = None,
+    ):
+        """Create a call to a function that implements requirement-like features, such as `record` and `terminate`.
```

Comment:
To avoid confusion, let's put "`terminate when`".

Reply:
Fixed this in 18a3e7202167a311d5fc80829585c82a98e24f27